### DISCUSSION TITLE
Issues compiling Rosetta 3.9 for RifDock

### DISCUSSION THREAD

**Discussion by maxrichardson250902** [RosettaCommons/rosetta#168] [Build & Install]
I am struggling to build Rosetta scripts using ninja build.py . I am using gcc ubuntu 5.3.1 and rosetta 3.9 downloaded from https://downloads.rosettacommons.org/downloads/academic/3.9/. I have downloaded the source version. 

When i run ./ninja_build.py cxx11_omp -t rosetta_scripts -remake. I get the following error

[max@max~R1rosetta_src_2018.09.60072.txt](https://github.com/user-attachments/files/17060188/max%40max.R1rosetta_src_2018.09.60072.txt)

**Comment by roccomoretti** [comment]
Looks like there's some issue with the compiler internals.  It looks like some part of the code generation is producing something the other part of the compiler can't understand.

```
{standard input}: Assembler messages:
{standard input}:24049: Warning: end of file not at end of a line; newline inserted
{standard input}:25163: Error: unknown pseudo-op: `.cfi_re'
{standard input}: Error: open CFI at the end of file; missing .cfi_endproc directive
g++-5: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
```

(When you get "internal compiler error" and "Please submit a full bug report, with preprocessed source", that indicates that there's something going on wrong internally with the compiler.)

You might be able to try compiling with a different compiler. A newer version or a different installation might fix it. -- for ninja_build.py, change your `CXX` and `CC` environment variables and then rebuild with the `-clean` flag. (CMake caches the compiler used, so you'll need a clean build to reset it.)

Alternatively, there may be something off with the OMP build. This isn't tested as much as the other builds, and it there may be some compiler/OMP difference that causes issues. You might want to try a non-OMP build, and see if that works. (The lack of OMP shouldn't affect things, except perhaps by making things slower, and even then only certain protocols would be affected.)

**Reply by maxrichardson250902** [reply]
Thank you for a detailed and quick reply! I'll try implementing these thanks.

**Reply by maxrichardson250902** [reply]
Hi, using GCC9 and running it clean, using the new BCOV/STABLE1 repository now. 
Still getting the errors from before.  It builds a varying amount objects each run. 

Could this be a memory issue?

![image](https://github.com/user-attachments/assets/0b9195ad-7d4e-425e-9b70-ef14949cec8d)

**Reply by maxrichardson250902** [reply]
Hi, its worked now.
I used Gcc9, BCOV/stable1, I removed the lines with REMOVE FOR RELEASE labels.
./ninja_build.py cxx11_omp -t rosetta_scripts -remake -clean -j7

not really sure why it worked this time and not the others
